app/components/Twitter.tsx

```diff
@@ -0,0 +1,210 @@
+import Vue from 'vue';
+import TsxComponent from 'components/tsx-component';
+import { Component, Prop, Watch } from 'vue-property-decorator';
+import { Inject } from 'services/core/injector';
+import { TwitterService } from 'services/integrations/twitter';
+import HFormGroup from 'components/shared/inputs/HFormGroup.vue';
+import { $t } from 'services/i18n';
+import { shell } from 'electron';
+import { Spinner, TextArea, Button } from 'streamlabs-beaker';
+import { ToggleInput } from 'components/shared/inputs/inputs';
+import { UserService } from 'services/user';
+
+import cx from 'classnames';
+import styles from './Twitter.m.less';
+
+@Component({})
+export class Twitter extends TsxComponent<{
+  streamTtitle: string;
+  midStreamMode: boolean;
+  updatingInfo: boolean;
+  value: string;
+}> {
+  @Inject() twitterService: TwitterService;
+  @Inject() userService: UserService;
+
+  @Prop() streamTitle: string;
+  @Prop() midStreamMode: boolean;
+  @Prop() updatingInfo: boolean;
+
+  priorTitle: string = '';
+  tweetModel: string = '';
+  shouldTweetModel: boolean = this.twitterService.state.tweetWhenGoingLive;
+
+  get isTwitch() {
+    return this.userService.platform.type === 'twitch';
+  }
+
+  get isPrime() {
+    return this.twitterService.state.prime;
+  }
+
+  get hasTwitter() {
+    return this.twitterService.state.linked;
+  }
+
+  get shouldTweet() {
+    return this.twitterService.state.tweetWhenGoingLive;
+  }
+
+  get twitterScreenName() {
+    return this.twitterService.state.screenName;
+  }
+
+  get csOnboardingComplete() {
+    return this.twitterService.state.creatorSiteOnboardingComplete;
+  }
+
+  get siteUrl() {
+    return this.twitterService.state.creatorSiteUrl;
+  }
+
+  get primeButtonText() {
+    return $t('Customize your URL');
+  }
+
+  get composeTweetText() {
+    return $t('Compose Tweet');
+  }
+
+  openPrime() {
+    shell.openExternal('https://streamlabs.com/editor/domain?ref=slobs_twitter&redirect=false');
+  }
+
+  linkTwitter() {
+    this.twitterService.openLinkTwitterDialog();
+  }
+
+  unlinkTwitter() {
+    this.twitterService.unlinkTwitter().then(() => this.getTwitterStatus());
+  }
+
+  setInitialTweetBody() {
+    let url = `${this.siteUrl}/home`;
+    if (!this.csOnboardingComplete && this.isTwitch) {
+      url = `https://twitch.tv/${this.userService.platform.username}`;
+    }
+    this.tweetModel = `${this.streamTitle} ${url}`;
+  }
+
+  async getTwitterStatus() {
+    await this.twitterService.getTwitterStatus();
+  }
+
+  async created() {
+    await this.getTwitterStatus();
+    this.priorTitle = this.streamTitle;
+    this.setInitialTweetBody();
+  }
+
+  updateTweetModel(tweet: string) {
+    this.tweetModel = tweet;
+  }
+
+  updateShouldTweetModel(shouldTweet: boolean) {
+    this.shouldTweetModel = shouldTweet;
+  }
+
+  @Watch('tweetModel')
+  onTweetChange(tweet: string) {
+    this.$emit('input', tweet);
```

Comment:
wondering why you're having an internal tweet model with a watcher instead of just emitting the changes

Reply:
Changed to passing strings around